### PR TITLE
Add navbar, headers, nav, about, splash and resources pages

### PR DESCRIPTION
**Pull request by mackwilson**

### CODE REVIEW

File: src/comps/PrivateNavbar.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import Navbar from 'react-bootstrap/Navbar';
+import Nav from 'react-bootstrap/Nav';
+import NavItem from 'react-bootstrap/NavItem';
```
Comment (lglof) [inline src/comps/PrivateNavbar.js:4]: you can pull these 3 lines into 1 with destructuring: 

`import { NavItem, Nav, Navbar } from "react-bootstrap";`

Comment (lglof) [reply src/comps/PrivateNavbar.js:4]: and I have no idea how to just do that w/ a code review. Will figure it out one day

File: src/comps/PublicNavbar.js [hunk 2]
```diff
@@ -1,15 +1,22 @@
 import React from 'react';
 import Navbar from 'react-bootstrap/Navbar';
 import Nav from 'react-bootstrap/Nav';
+import NavItem from 'react-bootstrap/NavItem';
```
Comment (lglof) [inline src/comps/PublicNavbar.js:4]: same 3 item destructuring: 

`import { NavItem, Nav, Navbar } from "react-bootstrap";`

File: src/css/App.css [hunk 3]
```diff
@@ -4,43 +4,193 @@ a {
...
+  margin-top: 2rem;
+}
+
+.gap-div {
+  height: 4rem;
 }
 
 .Menu a {
   color: #fff;
+  text-align: right !important;
```
Comment (lglof) [inline src/css/App.css:41]: do we need the `!important` tags in here?

File: src/css/App.css [hunk 4]
```diff
@@ -4,43 +4,193 @@ a {
...
 }
 
+.navbar {
+  align-items: right;
+  text-align: right !important;
+}
+
+.sosBtn a {
+  background-color: #fd79a8;
+  color: #343a40!important;
```
Comment (lglof) [inline src/css/App.css:55]: missing a space between colour and !important

File: src/css/App.css [hunk 5]
```diff
@@ -4,43 +4,193 @@ a {
...
+  font-weight: bold;
+  border: 1px solid #fff;
+  border-radius: 5px;
+  padding: 10px;
+}
+
+.sosBtn a:hover {
```
Comment (lglof) [inline src/css/App.css:62]: I love this hover thing. 
Worried abt the colours for now cause they're not v high contrast

File: src/css/App.css [hunk 6]
```diff
@@ -4,43 +4,193 @@ a {
...
+  margin-top: 8rem;
+}
+
+.BigHeader p {
+  font-size: calc(1vw + 1rem);
+  margin-top: 60px;
+  color: #fff;
+}
+
+.SectionHeader {
```
Comment (lglof) [inline src/css/App.css:142]: any way we can integrate SCSS or SASS into this so we don't have messy af css?

Comment (lglof) [reply src/css/App.css:142]: I know that by default vue's styling is in SASS, looks like there's a way to integrate: 

https://hugogiraudel.com/2015/06/18/styling-react-components-in-sass/

File: src/index.js [hunk 7]
```diff
@@ -1,27 +1,31 @@
...
-import HomePage from './pages/HomePage.js';
+import AboutPage from './pages/AboutPage.js';
 import ResourcePage from './pages/ResourcePage.js';
 import XpPage from './pages/XpPage.js';
 import SosPage from './pages/SosPage.js';
 import SignUpPage from './pages/SignUpPage.js';
 import LogInPage from './pages/LogInPage.js';
 import ProfilePage from './pages/ProfilePage.js';
+import BlogPage from './pages/BlogPage.js';
+import SplashPage from './pages/SplashPage.js';
```
Comment (lglof) [inline src/index.js:12]: this is a mess, would it be possible for us to do like an entrypoint to pages?
the entrypoint has all the pages defined as exports and then in index we just import the entrypoint? 

- it's possible that this is a lil complex for the structure we think we'll end up with

File: src/pages/ProfilePage.js [hunk 8]
```diff
@@ -1,11 +1,18 @@
 import React from 'react';
+import PrivateNavbar from '../comps/PrivateNavbar.js';
+import Footer from '../comps/Footer.js';
+import SmallHeader from '../comps/SmallHeader.js';
 import '../css/App.css';
```
Comment (lglof) [inline src/pages/ProfilePage.js:5]: I can give you dummy info for this based on some personas I've made

File: src/comps/PrivateNavbar.js (hunk 1 above)
Comment (mackwilson) [reply src/comps/PrivateNavbar.js:4]: If you do that, the entire react-bootstrap library will be imported. If you do it how I did it, you only import the specific components we want.

It has to do with how react-bootstrap library is structured. Each component in that library is its own export default class within its own file. This means you actually cant use the "{}" brackets at all, in fact - those are only for when there is more than one export on the file you are importing. You theoretically _could_ import all three in one line without the squiggly brackets, but as I said that will end up importing the whole library which is a no-no for build size. Its a massive library.

File: src/comps/PublicNavbar.js (hunk 2 above)
Comment (mackwilson) [reply src/comps/PublicNavbar.js:4]: Nope

File: src/css/App.css (hunk 3 above)
Comment (mackwilson) [reply src/css/App.css:41]: I only added the important tag when it was required to override default styling from react-bootstrap, so yes probably. I wouldn't add it if it wasn't necessary. 
I also didn't want to go through and undo every one of them to double check. You are welcome to if you want.

File: src/css/App.css (hunk 5 above)
Comment (mackwilson) [reply src/css/App.css:62]: Colours are not finalized, I'm just working on layout. We can change colours literally whenever we want

File: src/pages/ProfilePage.js (hunk 8 above)
Comment (mackwilson) [reply src/pages/ProfilePage.js:5]: Sure, but again, its a prototype of site layout. also the private pages are totally on the back burner - im prioritizing all public content first. When we start focussing on the private pages, then we will also have sign in flows that can be used to populate the profiles, too

File: src/index.js (hunk 7 above)
Comment (mackwilson) [reply src/index.js:12]: I dont know what you mean by entrypoint. If we make one page with all the "page" components exported, that page will be even more messy. 

More importantly, this is an index page. I'm sorry you find it "messy" (I dont), but the purpose is to import the page and then route to them. It only serves one purpose, and that purpose is very clear, and extremely standard for an `index` file. I think it is better to have everything clearly stated and straight forward. 

Also, this page is really only 33 lines - how can it be that bad when its not even a full screen length? Having a lot of imports doesnt automatically make a file problematic. This is also ALL of the pages we will ever have to route to, so this page is basically complete. 

I disagree that there is a problem.

File: src/css/App.css (hunk 6 above)
Comment (mackwilson) [reply src/css/App.css:142]: I can look into using sass, I have never used sass before personally which is why I didnt bother. Again, this is just the prototyping phase for the layout of the site, so I didn't think of it.

An alternative I considered was different css files for different pages for better organization. 

Also, I want to make the point that large css files are extremely common for large sites, and just something most developers deal with if css is what they are familiar with. I think this css file is fairly well-organized for its size. css always appears "messy". If you think this is "messy", you should see the css at most production companies lol

**Pull request merged**